rllib/utils/tf_utils.py

```diff
@@ -240,9 +240,11 @@ class for.
             assert tf1.executing_eagerly()
 
             from ray.rllib.policy.tf_policy import TFPolicy
+            from ray.rllib.policy.eager_tf_policy import EagerTFPolicy
 
-            # Create eager-class.
-            if hasattr(orig_cls, "as_eager"):
+            # Create eager-class (if not already one).
+            if hasattr(orig_cls, "as_eager") and \
+                    not issubclass(orig_cls, EagerTFPolicy):
                 cls = orig_cls.as_eager()
                 if config.get("eager_tracing"):
```

Comment:
per our offline message, I think the real fix should actually be to pull this
```
if config.get("eager_tracing"): cls = cls.with_tracing()
```
block out of as_eager block.
so we will try to enable tracing regardless if we as_eager() here, or somewhere else.
just FYI.

Reply:
You are right. Thanks for the catch.